engine/extended_providers/extended_providers.go

```diff
@@ -0,0 +1,164 @@
+package extendedproviders
```

Comment:
Better, but maybe `extproviders` or `xproviders` would be nice. Not a necessary change, but think about it.  If everyone if going to import it with an alias:
```go
import (
    xprov "github.com/filecoin-project/index-provider/engine/extended_provider"
)
```
Then maybe just change that package name to something shorter.


Another convention is that the package name is the base name of its source directory; the package in src/encoding/base64 is imported as "encoding/base64" but has name base64, not encoding_base64 and not encodingBase64.
-- Frrom: https://go.dev/doc/effective_go#package-names

Reply:
All great points, thanks! Changed and pushed.